packages/backend/src/models/ApplicationOptions.ts

```diff
@@ -0,0 +1,51 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { ContainerProviderConnection } from '@podman-desktop/api';
+import type { ModelInfo } from '@shared/models/IModelInfo';
+import type { Recipe } from '@shared/models/IRecipe';
+
+export type ApplicationOptions = ApplicationOptionsWithModelInference | ApplicationOptionsWithLlamaStack;
+
+interface ApplicationOptionsCommon {
+  inferModel: boolean;
+  startLlamaStack: boolean;
+  connection: ContainerProviderConnection;
+  recipe: Recipe;
+}
+
+export interface ApplicationOptionsWithModelInference extends ApplicationOptionsCommon {
+  inferModel: true;
+  model: ModelInfo;
+}
```

Comment:
```suggestion
type ApplicationOptionsWithModelInference = ApplicationOptionsCommon & {
  inferModel: true;
  model: ModelInfo;
}
```
Feel like this should be a type, not really fan of extending custom interface